Documentation/protocols/i2c.md

```diff
@@ -0,0 +1,175 @@
+# Enabling I2C on Raspberry Pi
+
+If you want to use the I2C capabilities of your Raspberry Pi, you will need to activate this feature. And if you want to run your code without elevated root privileged to access them, then, you may as well need to make couple of modifications. This tutorial is here to help you activating I2C and making sure you'll get the right permissions.
+
+## Basic Hardware I2C code
+
+The most simplest code you can build to use hardware I2C is the following (require C#9.0 to be activated):
+
+```csharp
+using System;
+using System.Device.I2c;
+
+Console.WriteLine("Hello I2C!");
+I2cDevice i2c = I2cDevice.Create(new I2cConnectionSettings(1, 0x12));
+i2c.WriteByte(0x42);
+var read = i2c.ReadByte();
+```
+
+In this case ```I2cDevice.Create(new I2cConnectionSettings(1, 0x12))``` will create an I2C device on the bus 1 (the default one) and with the device address 0x12.
+
+If you get an error message like this one, it means I2C has not been properly enabled:
+
+```text
+Hello I2C!
```

Comment:
This should be somewhere in the bottom in the Troubleshooting section or similar